### PR TITLE
chore: sync v3 with main

### PR DESCRIPTION
**Pull request by tswast**
Thought: maybe if we merge this as a merge commit, we can avoid some conflicts the next time we sync?

### CODE REVIEW

File: google/cloud/bigquery/table.py [hunk 1]
```diff
@@ -1690,6 +1688,8 @@ def to_arrow(
         """
         self._maybe_warn_max_results(bqstorage_client)
 
+        self._maybe_warn_max_results(bqstorage_client)
+
```
Comment (tswast) [inline google/cloud/bigquery/table.py:1692]: Duplicate

```suggestion
```

**Comment by tseaver** [comment]
@tswast I think the change to the samples image is [forthcoming soon](https://github.com/GoogleCloudPlatform/python-docs-samples/pull/6813).

File: owlbot.py [hunk 2]
```diff
@@ -32,8 +32,6 @@
     intersphinx_dependencies={
         "pandas": "http://pandas.pydata.org/pandas-docs/stable/",
         "geopandas": "https://geopandas.org/",
-        "proto-plus": ("https://proto-plus-python.readthedocs.io/en/latest/", None),
-        "protobuf": ("https://googleapis.dev/python/protobuf/latest/", None),
```
Comment (plamut) [inline owlbot.py:36]: IIRC these were added these ack when we re-added legacy proto-based types to `v3`. We should thus retain them.

File: owlbot.py [hunk 3]
```diff
@@ -55,10 +54,6 @@
     ],
 )
 
-# Remove unneeded intersphinx links, the library does not use any proto-generated code.
-s.replace("docs/conf.py", r'\s+"(proto-plus|protobuf)":.*$', "")
```
Comment (plamut) [inline owlbot.py:59]: Related to the comment above, we actually want to include proto* intersphinx links.

**Review by plamut — CHANGES_REQUESTED** [review]
It seems fine, apart from the intersphinx links that should be retained.

The unit tests still produce no warnings, and the `skipif` decorators in tests do not check for dependencies that have been made required in `v3` (BQ storage, pyarrow...) - these are the two main things that can sneak in without a merge conflict, as far as I can remember.

**Comment by plamut** [comment]
Added the `do not merge` label as an explicit reminder to use a _merge commit_ when merging this (do NOT squash). Hopefully the person who eventually merges this will notice this comment when seeking the reason why the label was added in the first place. :upside_down_face:

**Review by plamut — APPROVED** [review]

**Pull request merged**